wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserService.java

```
@@ -106,7 +104,10 @@ public Response create(@ApiParam("New user")
                                                        UnauthorizedException,
                                                        ConflictException,
                                                        ServerException {
-        userDto.withId(generate("user", ID_LENGTH));
+        if (userDto != null) {
+            //should be generated by userManager
```

may be if userDto == null or userDto.getId!= null we should throw exception instead of silently set null?

---

No. We always should set id to null because user should not be able to set id during creation and id should be generated by user manager.
Case when userDto is null then user entity should be fetched by token.
So we should not throw exception here